### ISSUE TITLE
 The db: pull command uses the same path on staging and local

### ISSUE THREAD

**Issue by nonzod** [sourcebroker/deployer-extended-database#18]
my config:

```php
...

host('local')
  ->hostname('localhost')
  ->set('deploy_path', getcwd())
  ->set('db_databases', [
    'database_default' => [
      get('local_db')
    ]
  ]);

host('staging')
  ->hostname(get('staging_host'))
  ->user('myuser') 
  ->port(22922)  
  ->set('deploy_path', get('staging_path'))
  ->set('branch', get('staging_branch'))
  ->set('db_databases', [
    'database_default' => [
      get('staging_db')
    ]
  ]);
```

at `./vendor/bin/dep db:pull staging -vvv`

```
[localhost] < ➤ Executing task db:download
[localhost] < [staging] > [ -d /<staging_path>/.dep/database/dumps ] || mkdir -p /<staging_path>/.dep/database/dumps
[localhost] < [localhost] > [ -d /<staging_path>/.dep/database/dumps ] || mkdir -p /<staging_path>/.dep/database/dumps
[localhost] < [localhost] < mkdir: impossibile creare la directory "/<staging_path>": Permission denied
```

Attempt to create the staging_path (**get('staging_path')**) directory locally instead of using local path (**getcwd()**).
The **.env** files are set in both environments with **INSTANCE="local"** for localhost and **INSTANCE="staging"** for staging.

**Comment by nonzod** [comment]
Found the problem.
Thanks to you suggestion I've noticed that `Configuration::getLocalHost()` use the **default_stage** value.
My **default_stage** is '**staging**', changing to '**local**' everythings works.
It's not intuitive but it works.

**Comment by kszymukowicz** [comment]
You should not change `default_stage` yourself. It is set automatically based on .env INSTANCE value!

Look on `vendor/sourcebroker/deployer-instance/deployer/config/set.php`

```
<?php

namespace Deployer;

use RuntimeException;
use SourceBroker\DeployerInstance\Env;

set('default_stage', function () {
    $env = new Env();
    $env->load();
    $instance = $env->get('INSTANCE');
    if ($instance === null) {
        throw new RuntimeException(
            'INSTANCE var is no set.',
            1602784218
        );
    }
    return $instance;
});

set('argument_stage', function () {
    return input()->getArgument('stage');
});
```

**Comment by nonzod** [comment]
Ok but this is in contrast with the official documentation:

https://deployer.org/docs/6.x/configuration#default_stage

**Comment by kszymukowicz** [comment]
I do not understand why is this in contrast. The `default_stage` is set for you based on .env data.

**Comment by kszymukowicz** [comment]
Ok - I see what you mean. 

In fact I never used "default_stage" in the meaning it was created for. 
I always do deploy with the given instance/stage.

So always "dep deploy prod" and never "dep deploy".

**Comment by kszymukowicz** [comment]
Without such trick most of commands would be much more complicated.

Examples:

- `dep db:export` would be `dep db:export local` when you run it local instance and `dep db:export staging` when you run it in staging instance.

- `dep db:pull live` would be `dep db:pull live --options=target:local` when you run it local instance and `dep db:pull live --options=target:staging` when you run it in staging instance.

So because I did not used "default_stage" for doing deploys I used it for simplifying database operations.
